### DISCUSSION TITLE
Dodatkowy atrybut w metodzie order/checkout-forms/{id} z uwagi na Allegro Pay

### DISCUSSION THREAD

**Discussion by ricarderobl** [allegro/allegro-api#7460] [Suggestions - resolved]
**Skoro jest doliczana opłata dla sprzedawców w wysokości 0,43% za Allegro Pay, to ta informacja powinna być jawnie dostępna w tej metodzie jako atrybut "AllegroPay": bool.** Wtedy będzie można zweryfikować czy opłaty na billingu pokrywają się ze złożonymi zamówieniami.

W przeciwnym razie można doliczać sobie do każdego zamówienia wedle woli tę opłatę a sprzedawcy nie mają żadnych narzędzi masowej weryfikacji czy ta opłata była słusznie naliczona, gdyż nijak nie da się aktualnie zobaczyć w REST jakie zamówienia są składane poprzez metodę Allegro Pay.

Uprzedzając odpowiedź osób odpowiedzialnych za ten bałagan (jak można wprowadzać opłatę bez możliwości jej jakiejkolwiek weryfikacji?) to billing nie jest miejscem do weryfikacji, a rachunkiem. Czyli najpierw jest wykonywana usługa X i wtedy jest wystawiany rachunek Y. A nie jest wystawiany rachunek Y by dowiedzieć się, że wykonano usługę X.

Przykładowo zamawiając usługę zmiany opon wiemy co zlecamy i na tej podstawie otrzymujemy rachunek za zmianę opon. **Aktualnie natomiast Allegro zrobiło, że samochód trafia na warsztat, nie ma zlecenia jakie naprawy należy wykonać, a Allegro wystawia rachunek i mówi, że w samochodzie była zmieniana żarówka.** Jak niby wtedy jest to poprawne, skoro nie jesteśmy w stanie sprawdzić, czy faktycznie zlecaliśmy wymianę żarówki?

**Comment by hahyes** [comment]
@ricarderobi Oczywiście, że to powinno być już dawno. Niestety REST API Allegro nigdy nie było i raczej nigdy nie będzie kompletne pod względem informacyjnym. Część funkcjonalności z frontu nigdy nie trafiła do API - np. pobieranie formularzy zwrotu. Trzeba używać scrappera, żeby pobrać cholernego PDFa. Chcesz pobrać listę przesyłek i zsynchronizować dane, ponieważ jakiś niedożywiony pracownik postanowił nadać przesyłkę ręcznie? Nie da się. Chcesz pobrać etykietę ostatniej przesyłki, ale nie zapisałeś `parcelId`? Sorry bucko, you can't.

O nieodtwarzalnych błędach takich jak puste wyniki wyszukiwania ofert na koncie sprzedawcy już nie wspomnę, bo wczoraj myślałem że oszaleje z tym. API sobie zwróciło zamiast ofert pustą kolekcję. Gdy API zwraca pustą kolekcję to się trzeba najlepiej upewnić, że faktycznie tego tam nie ma. Absurd.

**Comment by PrzemyslawLukanowski** [comment]
Zaktualizowaliśmy status sugestii jako "wdrożona". W polach `payment.features `i `surcharges.features` [zwracamy informację o Allegro Pay](https://developer.allegro.pl/news/zamowienia-dodalismy-dodatkowe-informacje-o-platnosci-wwxoRvvxET5).
